src/graph/transform/cuda/cuda_to_block.cu

```diff
@@ -137,209 +137,74 @@ class DeviceNodeMapMaker {
   IdType max_num_nodes_;
 };
 
-// Since partial specialization is not allowed for functions, use this as an
-// intermediate for ToBlock where XPU = kDGLCUDA.
 template <typename IdType>
-std::tuple<HeteroGraphPtr, std::vector<IdArray>> ToBlockGPU(
-    HeteroGraphPtr graph, const std::vector<IdArray>& rhs_nodes,
-    const bool include_rhs_in_lhs, std::vector<IdArray>* const lhs_nodes_ptr) {
-  std::vector<IdArray>& lhs_nodes = *lhs_nodes_ptr;
-  const bool generate_lhs_nodes = lhs_nodes.empty();
-
-  const auto& ctx = graph->Context();
-  auto device = runtime::DeviceAPI::Get(ctx);
-  cudaStream_t stream = runtime::getCurrentCUDAStream();
-
-  CHECK_EQ(ctx.device_type, kDGLCUDA);
-  for (const auto& nodes : rhs_nodes) {
-    CHECK_EQ(ctx.device_type, nodes->ctx.device_type);
-  }
-
-  // Since DST nodes are included in SRC nodes, a common requirement is to fetch
-  // the DST node features from the SRC nodes features. To avoid expensive
-  // sparse lookup, the function assures that the DST nodes in both SRC and DST
-  // sets have the same ids. As a result, given the node feature tensor ``X`` of
-  // type ``utype``, the following code finds the corresponding DST node
-  // features of type ``vtype``:
-
-  const int64_t num_etypes = graph->NumEdgeTypes();
-  const int64_t num_ntypes = graph->NumVertexTypes();
-
-  CHECK(rhs_nodes.size() == static_cast<size_t>(num_ntypes))
-      << "rhs_nodes not given for every node type";
-
-  std::vector<EdgeArray> edge_arrays(num_etypes);
-  for (int64_t etype = 0; etype < num_etypes; ++etype) {
-    const auto src_dst_types = graph->GetEndpointTypes(etype);
-    const dgl_type_t dsttype = src_dst_types.second;
-    if (!aten::IsNullArray(rhs_nodes[dsttype])) {
-      edge_arrays[etype] = graph->Edges(etype);
-    }
-  }
-
-  // count lhs and rhs nodes
-  std::vector<int64_t> maxNodesPerType(num_ntypes * 2, 0);
-  for (int64_t ntype = 0; ntype < num_ntypes; ++ntype) {
-    maxNodesPerType[ntype + num_ntypes] += rhs_nodes[ntype]->shape[0];
-
+struct GetMappingIdsGPU {
+  std::tuple<std::vector<IdArray>, std::vector<IdArray>> operator()(
+      const HeteroGraphPtr& graph, bool include_rhs_in_lhs, int64_t num_ntypes,
+      const DGLContext& ctx, const std::vector<int64_t>& maxNodesPerType,
+      const std::vector<EdgeArray>& edge_arrays,
+      const std::vector<IdArray>& src_nodes,
+      const std::vector<IdArray>& rhs_nodes,
+      std::vector<IdArray>* const lhs_nodes_ptr,
+      std::vector<int64_t>* const num_nodes_per_type_ptr) {
+    std::vector<IdArray>& lhs_nodes = *lhs_nodes_ptr;
+    std::vector<int64_t>& num_nodes_per_type = *num_nodes_per_type_ptr;
+    const bool generate_lhs_nodes = lhs_nodes.empty();
+    auto device = runtime::DeviceAPI::Get(ctx);
+    cudaStream_t stream = runtime::getCurrentCUDAStream();
+
+    // allocate space for map creation process
```

Comment:
Capitalize the first char and end with dot.
For all other comments.

Reply:
fixed